platform/x11/os_x11.cpp

```diff
@@ -1743,7 +1743,7 @@ void OS_X11::handle_key_event(XKeyEvent *p_event, bool p_echo) {
 			// is correct, but the xorg developers are
 			// not very helpful today.
 
-			::Time tresh = ABS(peek_event.xkey.time - xkeyevent->time);
+			::Time tresh = peek_event.xkey.time - xkeyevent->time;
```

Comment:
Are we sure that `peek_event.xkey.time > xkeyevent->time` at all times?

Reply:
The original `ABS` macro wasn't quite correct due to the types involved being unsigned. The `ABSDIFF` macro defined in `servers/audio/reverb_sw.cpp` does the correct thing for unsigned integers. Perhaps move that macro into `core/typedefs.h` alongside `ABS` and use it here?